File: frontend/src/types/schema-matrix.ts
export interface FieldPipelineInfo {
  pipeline_id: string;
  pipeline_name: string;
}

export interface FieldFrequencyRow {
  field_name: string;
  frequency: number;
  pipelines: FieldPipelineInfo[];
}

export interface SchemaMatrixResponse {
  fields: FieldFrequencyRow[];
}

File: backend/app/schemas/schema_matrix.py
from pydantic import BaseModel


class FieldPipelineInfo(BaseModel):
    pipeline_id: str
    pipeline_name: str


class FieldFrequencyRow(BaseModel):
    field_name: str
    frequency: int
    pipelines: list[FieldPipelineInfo]


class SchemaMatrixResponse(BaseModel):
    fields: list[FieldFrequencyRow]
